4. LINQ/Task/Task/Task/LinqSamples.cs

```diff
@@ -0,0 +1,272 @@
+				from num in numbers
+				where num < 5
+				select num;
+
+			Console.WriteLine("Numbers < 5:");
+			foreach (var x in lowNums)
+			{
+				Console.WriteLine(x);
+			}
+		}
+
+		[Category("Restriction Operators")]
+		[Title("Where - Task 2")]
+		[Description("This sample return return all presented in market products")]
+
+		public void Linq2()
+		{
+			var products =
+				from p in dataSource.Products
+				where p.UnitsInStock > 0
+				select p;
+
+			foreach (var p in products)
+			{
+				ObjectDumper.Write(p);
+			}
+		}
+
+		[Category("Task")]
+		[Title("Task 1")]
+		[Description("Get a list of all customers whose total turnover (the sum of all orders) exceeds some value of X")]
+		public void Linq001()
+		{
+			decimal x = 20000;
+
+			var customers = dataSource.Customers.Where(c => c.Orders.Sum(o => o.Total) > x);
+
+			foreach (var customer in customers)
+			{
+				ObjectDumper.Write($"{customer.CustomerID} {customer.CompanyName} {customer.Orders.Sum(o => o.Total)}");
+			}
+		}
+
+		[Category("Task")]
+		[Title("Task 2")]
+		[Description("Make a list of suppliers located in the same country and the same city for each client")]
+		public void Linq002()
+		{
+			var customers = dataSource.Customers
+				.Select(c => new
+				{
+					c.CustomerID,
+					c.Country,
+					c.City,
+					Suppliers = dataSource.Suppliers.Where(s => s.Country == c.Country && s.City == c.City)
+				});
+
+			foreach (var customer in customers)
+			{
+				foreach (var supplier in customer.Suppliers)
+				{
+					ObjectDumper.Write($"{customer.CustomerID} {customer.Country} {customer.City} {supplier.SupplierName} {supplier.Country} {supplier.City} ");
+				}
+			}
+		}
+
+		[Category("Task")]
+		[Title("Task 3")]
+		[Description("Find all customers who have orders that exceed the sum of X")]
+		public void Linq003()
+		{
+			decimal x = 10000;
+
+			var customers = dataSource.Customers.Where(c => c.Orders.Any(o => o.Total > x));
+
+			foreach (var customer in customers)
+			{
+				ObjectDumper.Write($"{customer.CustomerID} {customer.CompanyName} {customer.Orders.Max(o => o.Total)}");
+			}
+		}
+
+		[Category("Task")]
+		[Title("Task 4")]
+		[Description("Get a list of customers including the month of the year they became clients (consider client first order month as a required date)")]
+		public void Linq004()
+		{
+			var customers = dataSource.Customers
+				.Select(c => new
+				{
+					c.CustomerID,
+					c.CompanyName,
+					FirstOrderDate = c.Orders.OrderBy(o => o.OrderDate).Select(o => o.OrderDate).FirstOrDefault()
+				}
+				);
+
+			foreach (var customer in customers)
+			{
+				ObjectDumper.Write($"{customer.CustomerID} {customer.CompanyName} {customer.FirstOrderDate}");
+			}
+		}
+
+		[Category("Task")]
+		[Title("Task 5")]
+		[Description("Do the previous task, but get the list sorted by year, month, customer turnover (from the maximum to the minimum), and the client's name")]
+		public void Linq005()
+		{
+			var customers = dataSource.Customers
```

Comment:
will this query work correctly for the customer without orders?

Reply:
Fixed